### PR TITLE
Two small man page enhancements

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: man/bootup.xml [hunk 1]
```diff
@@ -23,17 +23,25 @@
...
 
-    <para>A number of different components are involved in the system
-    boot. Immediately after power-up, the system BIOS will do minimal
-    hardware initialization, and hand control over to a boot loader
-    stored on a persistent storage device. This boot loader will then
-    invoke an OS kernel from disk (or the network). In the Linux case,
-    this kernel (optionally) extracts and executes an initial RAM disk
-    image (initrd), such as generated by
+    <para>A number of different components are involved in the boot of a Linux system. Immediately after
+    power-up, the system BIOS will do minimal hardware initialization, and hand control over to a boot loader
```
Comment (poettering) [inline man/bootup.xml]: if you touch that, we probably shouldn't call this "system BIOS" anymore, but "system firmware"...

File: man/bootup.xml [hunk 2]
```diff
@@ -23,17 +23,25 @@
...
+    (e.g. <ulink url="https://www.gnu.org/software/grub/">GRUB</ulink> or
+    <citerefentry><refentrytitle>systemd-boot</refentrytitle><manvolnum>7</manvolnum></citerefentry>)
```
Comment (poettering) [inline man/bootup.xml]: please swap the two, please, let's name our own stuff first ;-)

File: man/bootup.xml [hunk 3]
```diff
@@ -23,17 +23,25 @@
...
+    stored on a persistent storage device. This boot loader will then invoke an OS kernel from disk (or the
+    network). On systems using EFI or other types of firmware, this firmware may also load the kernel
+    directly.</para>
+
+    <para>The kernel (optionally) mounts an in-memory file system, such as generated by
     <citerefentry project='die-net'><refentrytitle>dracut</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
-    which looks for the root file system (possibly using
-    <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>
-    for this). After the root file system is found and mounted, the
+    which looks for the root file system. Initramfs — an lightweight read-only in-memory file system based on
```
Comment (poettering) [inline man/bootup.xml]: "an lightweight" → "a lightweight"

File: man/bootup.xml [hunk 4]
```diff
@@ -23,17 +23,25 @@
...
-    image (initrd), such as generated by
+    <para>A number of different components are involved in the boot of a Linux system. Immediately after
+    power-up, the system BIOS will do minimal hardware initialization, and hand control over to a boot loader
+    (e.g. <ulink url="https://www.gnu.org/software/grub/">GRUB</ulink> or
+    <citerefentry><refentrytitle>systemd-boot</refentrytitle><manvolnum>7</manvolnum></citerefentry>)
+    stored on a persistent storage device. This boot loader will then invoke an OS kernel from disk (or the
+    network). On systems using EFI or other types of firmware, this firmware may also load the kernel
+    directly.</para>
+
+    <para>The kernel (optionally) mounts an in-memory file system, such as generated by
```
Comment (poettering) [inline man/bootup.xml]: such as → commonly? frequently? typically?

File: man/bootup.xml [hunk 5]
```diff
@@ -23,17 +23,25 @@
...
     <citerefentry project='die-net'><refentrytitle>dracut</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
-    which looks for the root file system (possibly using
-    <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>
-    for this). After the root file system is found and mounted, the
+    which looks for the root file system. Initramfs — an lightweight read-only in-memory file system based on
+    tmpfs is usually used for this, but in the past normal file systems using an in-memory block device
+    (ramdisk) were used, and the name "initrd" is still used to describe both mechanisms. It's the boot loader
```
Comment (poettering) [inline man/bootup.xml]: mechanisms → concepts?

File: man/bootup.xml [hunk 6]
```diff
@@ -23,17 +23,25 @@
...
+    or the firmware that loads both the kernel and initrd/initramfs images into memory, but the kernel which
+    interprets it as a file system.
+    <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry> may be used
+    to manage services in the initrd, similarly to the real system.</para>
```
Comment (poettering) [inline man/bootup.xml]: so, strictly speaking initrds are not a file system but cpio archives which are unpacked by the kernel early on into a file system. When we explain the concept we should probably get this right.

File: man/systemd.service.xml [hunk 7]
```diff
@@ -473,14 +473,18 @@
...
-        <varname>ExecStopPost=</varname> are executed during a service restart operation.</para>
-
-        <para>It is recommended to use this setting for commands that communicate with the service requesting clean
-        termination. When the commands specified with this option are executed it should be assumed that the service is
-        still fully up and is able to react correctly to all commands. For post-mortem clean-up steps use
-        <varname>ExecStopPost=</varname> instead.</para></listitem>
+        service failed to start up correctly and is shut down again. Also note that the stop operation is always
+        performed if the service started successfully, even if the processes in the service terminated on their
+        own or were killed. The stop commands must be prepared to deal with that case. <varname>$MAINPID</varname>
+        will be empty if systemd knows that the main process exited by the time the stop commands are called.</para>
```
Comment (poettering) [inline man/systemd.service.xml]: empty? unset!

File: man/bootup.xml (hunk 6 above)
Comment (keszybz) [reply man/bootup.xml]: Yikes, I misunderstood how the initramfs works. I updated the description.

**Comment by keszybz** [comment]
Updated with the requested changes and a fix for the NEWS patch that went in in the meantime.

**Pull request merged**